### PR TITLE
fix(curriculum): Regex fix for Create a Variable for a Specific Area

### LINKED ISSUE #41500: curriculum: Bug in the Create a Variable for a Specific Area challenge
**Describe your problem and how to reproduce it:**

The challenge expects the camper to add the following CSS property to the `.penguin` class:

```css
--penguin-belly: white;
```

However, the following code (which does not accomplish the challenge) passes due to a lack of specificity in the regex assertion:

```css
background-color: --penguin-belly:white;
```

Note that this code will not change the colour of the penguin's belly and should not pass.

This could potentially be fixed with a change in the test regex to:

```js
/.penguin\s*?{[\s\S]*\n\s*--penguin-belly\s*?:\s*?white\s*?;[\s\S]*}/gi
```

which would assert the `--penguin-belly` is preceded by a new line and some spaces, but no non-space characters.

**Add a Link to the page with the problem:**

https://www.freecodecamp.org/learn/responsive-web-design/basic-css/change-a-variable-for-a-specific-area

**If possible, add a screenshot here (you can drag and drop, png, jpg, gif, etc. in this box):**

image

Screenshot provided by `Zeal#6159` on Discord.

### PR DESCRIPTION
**Pull request by Sembauke**
Checklist:

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] All the files I changed are in the same world language, for example: only English changes, or only Chinese changes, etc.

Closes #41500
these are the suggested changes made by Nhcarrigan.

### CODE REVIEW

**Comment by Sembauke** [comment]
I was wondering if this should be added to the Chinese and Spanish versions too?

**Comment by RandellDawson** [comment]
@Sembauke Your changes are causing the test to fail.  You should run `npm run test:curriculum` locally before submitting new changes, to confirm all the tests are passing.

**Comment by nhcarrigan** [comment]
Ah shoot, this is partly my fault. I suggested a potential regex change that would match the camper's input based on the seed, but the solution we test against is minified a bit and won't pass:

image

File: curriculum/challenges/english/01-responsive-web-design/basic-css/change-a-variable-for-a-specific-area.md [hunk 1]
```diff
@@ -23,7 +23,7 @@ The `penguin` class should reassign the `--penguin-belly` variable to `white`.
 
 ```js
 assert(
-  code.match(/.penguin\s*?{[\s\S]*\n\s*--penguin-belly\s*?:\s*?white\s*?;[\s\S]*}/gi)
+  code.match(/\.penguin\s*?{[\s\S]*(?<!background-color: )--penguin-belly\s*?:\s?white\s*?;[\s\S]*}/gi)
```
Comment (RandellDawson) [inline curriculum/challenges/english/01-responsive-web-design/basic-css/change-a-variable-for-a-specific-area.md:26]: Not sure if negative lookbehinds are supported across all the browsers we support currently.  I am also not sure if Babel will transform it into something that can work.  We have no other challenges in the curriculum that use a negative lookbehind in a challenge test.

Comment (Sembauke) [reply curriculum/challenges/english/01-responsive-web-design/basic-css/change-a-variable-for-a-specific-area.md:26]: Maybe another test won't hurt:

your `.penguin` class should not contain `background-color`

Test:
`code.match(/^((?!background-color:).)*$/gs)`

Comment (Sembauke) [reply curriculum/challenges/english/01-responsive-web-design/basic-css/change-a-variable-for-a-specific-area.md:26]: @RandellDawson

**Comment by Sembauke** [comment]
Maybe another test won't hurt:

your  `.penguin class`  should not contain `background-color`

Test:
`code.match(/^((?!background-color\s*?:\s*?)[\s\S])*$/g)`

File: curriculum/challenges/english/01-responsive-web-design/basic-css/change-a-variable-for-a-specific-area.md (hunk 1 above)
Comment (ojeytonwilliams) [reply curriculum/challenges/english/01-responsive-web-design/basic-css/change-a-variable-for-a-specific-area.md:26]: We currently only support Firefox and Chrome (both of which can [handle it](https://caniuse.com/?search=lookbehind)).  As far as Babel goes, we don't do anything to the tests themselves.

Comment (Sembauke) [reply curriculum/challenges/english/01-responsive-web-design/basic-css/change-a-variable-for-a-specific-area.md:26]: This is a solution that also works, but I've also made another suggestion.

**Comment by raisedadead** [comment]
The last few comments are non-conclusive on the next steps with this PR. @RandellDawson @ojeytonwilliams does this need further changes or is this good to go?

**Review by hassaanp — APPROVED** [review]

**Review by ojeytonwilliams — APPROVED** [review]
LGTM :+1: 

As discussed, there shouldn't be a problem for the browsers we support and the solution looks good.

**Review by raisedadead — APPROVED** [review]

**Pull request merged**